Tools/scripts/install-prereqs-arch.sh

```diff
@@ -65,6 +65,11 @@ if ! grep -Fxq "$exportline" ~/.bashrc ; then
     fi
 fi
 
+# Get the correct ardupilot path if we are running the script inside the repository
+if git rev-parse --show-toplevel 2> /dev/null; then
+    CWD=$(git rev-parse --show-toplevel)
+fi
```

Comment:
Almost at the end we get the script directory, wouldn't that be better to use? At least as an else clause (I'm guessing it's unlikely you are in one git repo running a script from another - in which case the code above would be a problem -, but being at a level up the ArduPilot git repo is probably likely as you would have just cloned).